aliyun/src/main/java/org/apache/iceberg/aliyun/oss/BaseOSSFile.java

```diff
@@ -23,18 +23,25 @@
 import com.aliyun.oss.OSSErrorCode;
 import com.aliyun.oss.OSSException;
 import com.aliyun.oss.model.SimplifiedObjectMeta;
+import org.apache.iceberg.aliyun.AliyunProperties;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
 
 abstract class BaseOSSFile {
   private final OSS client;
   private final OSSURI uri;
+  private AliyunProperties aliyunProperties;
   private SimplifiedObjectMeta metadata;
 
   BaseOSSFile(OSS client, OSSURI uri) {
     this.client = client;
     this.uri = uri;
   }
 
+  BaseOSSFile(OSS client, OSSURI uri, AliyunProperties aliyunProperties) {
+    this(client, uri);
```

Comment:
Instead of initializing this constructor on top of the `BaseOSSFile(OSS client, OSSURI uri)`,  I will prefer to initialize the `BaseOSSFile(OSS client, OSSURI uri)` on top of this constructor. In the current approach,  the `aliyunProperties` will be set as a `null`, then we have to check whether it is nullable everywhere in the upper classes such as `OSSOutputStream`. 

I think the aliyunProperties can be initialized as a `new AliyunProperties()` by default.